### DISCUSSION TITLE
Enhancement for IAsyncEnumerable

### DISCUSSION THREAD

**Discussion by ArnaudDebaene** [JSkimming/Castle.Core.AsyncInterceptor#182] [Ideas]
Hello,

Thanks for the great library.

I've written an "enhanced" version of your library to be able to handle methods that returns IAsyncEnumerable<T>.

Basically, I've added a 4th method to the IAsyncInterceptor interface : 
`void InterceptAsyncEnumerable<TResult>(IInvocation invocation);`
...and I've enriched ProcessingAsynInterceptor and AsyncDeterminationInterceptor to handle these calls.

Would you be interested with this? I can push my code into a pr if you want.

**Comment by JSkimming** [comment]
Hi @ArnaudDebaene, I'm interested to see what it looks like. Feel free to submit a PR; we can use that as the discussion point.

**Reply by ArnaudDebaene** [reply]
Done : https://github.com/JSkimming/Castle.Core.AsyncInterceptor/pull/183
